sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientBuilderTest.java

```diff
@@ -83,6 +84,25 @@ public void emptyConnectionString() {
         });
     }
 
+    @Test
+    @DoNotRecord
+    public void nullCredentials() {
+        assertThrows(NullPointerException.class, () -> {
+            final ConfigurationClientBuilder builder = new ConfigurationClientBuilder();
+            builder.buildClient();
+        });
+    }
+
+    @Test
+    @DoNotRecord
+    public void multipleCredentialsExist() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            final ConfigurationClientBuilder builder = new ConfigurationClientBuilder();
+            builder.connectionString(FAKE_CONNECTION_STRING)
+                .credential(new DefaultAzureCredentialBuilder().build()).buildClient();
```

Comment:
For non recorded tests, we don't need to use DefaultAzureCredential. We can use a fake token credential.
```suggestion
           TokenCredential credentials = request -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX))
            builder.connectionString(FAKE_CONNECTION_STRING)
                .credential(credentials).buildClient();
```